unified-codes/apps/data-service/src/app/v2/types/entity/ProductEntity.ts

```diff
@@ -0,0 +1,42 @@
+import { EEntityType, IEntity, Entity } from './Entity';
+import { IRouteEntity, RouteEntity } from './RouteEntity';
+import { IProperty, Property } from '../Property/Property';
+
+export enum EProductField {
+    Code = 'code',
+    Name = 'name',
+    Type = 'type',
+    Combines = 'combines',
+    Children = 'children',
+    Properties = 'properties',
+};
+
+export type IProductEntityChild = IRouteEntity;
+export type IProductEntityChildren = IProductEntityChild[];
+
+export type IProductEntityProperty = IProperty;
+export type IProductEntityProperties = IProductEntityProperty[];
+
+export interface IProductEntity extends IEntity {
+  type: EEntityType.Product;
+  combines?: IProductEntity[];
+  children?: IProductEntityChildren;
+  properties?: IProductEntityProperties;
+}
+
+export class ProductEntity extends Entity implements IProductEntity {
+  readonly type: EEntityType.Product;
+  readonly combines?: IProductEntity[];
+  readonly children?: IProductEntityChildren;
+  readonly properties?: IProductEntityProperties;
+
+  constructor(product: IProductEntity) {
+    super(product);
+    this.type = EEntityType.Product;
+    this.combines = product.combines?.map((sibling: IProductEntity) => new ProductEntity(sibling));
+    this.children = product.children?.map((child: IProductEntityChild) => new RouteEntity(child));
```

Comment:
Just wondering about some products, as it seems some don't have a route defined (i.e. `ProductEntity` links straight to `DoseUnit`. One example of this is `Examination Glove`). Would we use an empty `RouteEntity` or allow product to have `DoseUnit` children?